modules/accredited_representative_portal/app/services/accredited_representative_portal/power_of_attorney_request_service/accept.rb

```diff
@@ -0,0 +1,137 @@
+      attr_accessor :form_submission
+
+      def initialize(poa_request, creator, reason)
+        @poa_request = poa_request
+        @creator = creator
+        @reason = reason
+      end
+
+      def call
+        ApplicationRecord.transaction do
+          create_resolution!
+          create_form_submission!
+        end
+
+        service = BenefitsClaims::Service.new(poa_request.claimant.icn)
+        response = service.submit2122(attributes)
+
+        form_submission.update(
+          service_id: response.body.dig('data', 'id'),
+          service_response: response.body.to_json
+        )
+        form_submission
+      # TODO: call PowerOfAttorneyFormSubmissionJob.perform_async(poa_form_submission)
+      # Transient 5xx errors: delete objects created, re-raise error
+      rescue *transient_error_types => e
+        form_submission.delete
+        resolution.delete
+        raise e
+      # All other errors: save error data on form submission
+      rescue => e
+        create_error_submission(e.message)
+        raise e
+      end
+
+      def create_resolution!
+        resolving = PowerOfAttorneyRequestDecision.create!(
+          type: PowerOfAttorneyRequestDecision::Types::ACCEPTANCE, creator:
+        )
+        ##
+        # This form triggers the uniqueness validation, while the
+        # `@poa_request.create_resolution!` form triggers a more obscure
+        # `RecordNotSaved` error that is less functional for getting
+        # validation errors.
+        #
+        @resolution = PowerOfAttorneyRequestResolution.create!(
+          power_of_attorney_request: poa_request,
+          resolving:,
+          reason:
+        )
+      end
+
+      def create_form_submission!
+        @form_submission = PowerOfAttorneyFormSubmission.create!(
+          power_of_attorney_request: poa_request,
+          status: :enqueue_succeeded,
+          status_updated_at: DateTime.current
+        )
+      end
+
+      def create_error_submission(message)
+        PowerOfAttorneyFormSubmission.create(
+          service_response: message,
+          error_message: message,
+          status: :enqueue_failed,
+          status_updated_at: DateTime.current
+        )
+      end
+
+      def attributes
+        {}.tap do |a|
+          a[:veteran] = veteran_data
+          a[:serviceOrganization] = service_org_data
+          a[:recordConsent] = form_data.dig('authorizations', 'recordDisclosure')
+          a[:consentLimits] = form_data.dig('authorizations', 'recordDisclosureLimitations')
+          a[:consentAddressChange] = form_data.dig('authorizations', 'addressChange')
+        end
+      end
+
+      def service_org_data
+        {
+          poaCode: poa_request.power_of_attorney_holder_poa_code,
+          registrationNumber: poa_request.accredited_individual_registration_number
+        }
+      end
+
+      def form_data
+        @form_data ||= JSON.parse(poa_request.power_of_attorney_form.data)
```

Comment:
`PowerOfAttorneyForm#parsed_data` is already this method implementation and it is memoized so you can call it repeatedly too.